maestro/demos/agents/crewai/generic/generic_agent.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+"""
+This script uses the crew.ai framework to find activities suitable for cold or wet weather. 
+
+Dependencies:
+- crewai
+
+Usage:
+- Ensure ollama is available on localhost:11434 and the llama3.1 model is available.
+"""
+
+import os
+from crewai import Agent, Crew, Task, Process, LLM
+from crewai.project import CrewBase, agent, task, crew
+from crewai.tools import tool
+
+@CrewBase
+class Generic_Crew:
+    """
+    Defines a class to manage a crew that finds activities to do in cold or wet weather.
+    """
+
+    # TODO Set model/URL from configuration
+    llm = LLM(
+        model=os.getenv("MAESTRO_DEMO_OLLAMA_MODEL", "ollama/llama3.1"),
+        base_url=os.getenv("MAESTRO_DEMO_OLLAMA_URL", "http://localhost:11434")
```

Comment:
Same thing. We could add `model_provider` in the agent YAML for this. OR a structure like:
```yaml
...
model_config: 
    name: "name-of-model"
    provider: "URL of the provider"
    ...
```

If we can make `model` be a string and optionally an object then no need for `model_config` but otherwise this could do it.

WDYT?

Reply:
Same as above.